### PR TITLE
cherry-picking trivial_forwarding support

### PR DESCRIPTION
**Pull request by jjsjann123**
cherry-picks this one https://github.com/csarofeen/pytorch/pull/1995

### CODE REVIEW

**Comment by jjsjann123** [comment]
Will update once I figure out the IMA in the original PR.

File: torch/csrc/jit/codegen/cuda/kernel_cache.cpp [hunk 1]
```diff
@@ -621,11 +621,16 @@ std::vector<at::Tensor> FusionKernelRuntime::runWithInput(
...
   }
 
+  // group should share cache id.
+  auto group_cache_id = args.getCacheId();
   for (auto group_to_run : runtime_workspace_.group_run_order) {
     // TODO: index mode should be updated per segmented kernel
     // Prepare input vector
     KernelArgumentHolder group_runtime_inputs(args.getIndexMode());
     group_runtime_inputs.setDeviceIndex(args.getDeviceIndex());
+    if (group_cache_id.has_value()) {
```
Comment (jjsjann123) [inline torch/csrc/jit/codegen/cuda/kernel_cache.cpp]: Oops, our cache seems to be broken. I'll put this in a separate PR and track it there.

**Comment by IvanYashchuk** [comment]
Thank you for working on this! ~Let's merge when https://github.com/csarofeen/pytorch/pull/1995 is finalized.~ I'm going to merge this one, as it's good enough for this branch.

**Comment by IvanYashchuk** [comment]
With this PR tests for "copy_to" pass.

**Pull request merged**